packages/message-box/src/index.vue

```diff
@@ -0,0 +1,354 @@
+          >
+            <i class="el-message-box__close el-icon-close"></i>
+          </button>
+        </div>
+        <div class="el-message-box__content">
+          <div class="el-message-box__container">
+            <div
+              v-if="icon && !center && message !== ''"
+              :class="['el-message-box__status', icon]"
+            >
+            </div>
+            <div v-if="message !== ''" class="el-message-box__message">
+              <slot>
+                <p v-if="!dangerouslyUseHTMLString">{{ message }}</p>
+                <p v-else v-html="message"></p>
+              </slot>
+            </div>
+          </div>
+          <div v-show="showInput" class="el-message-box__input">
+            <el-input
+              ref="input"
+              v-model="inputValue"
+              :type="inputType"
+              :placeholder="inputPlaceholder"
+              @keydown.enter="handleInputEnter"
+            />
+            <div class="el-message-box__errormsg" :style="{ visibility: !!editorErrorMessage ? 'visible' : 'hidden' }">{{ editorErrorMessage }}</div>
+          </div>
+        </div>
+        <div class="el-message-box__btns">
+          <el-button
+            v-if="showCancelButton"
+            :loading="cancelButtonLoading"
+            :class="[ cancelButtonClasses ]"
+            :round="roundButton"
+            size="small"
+            @click="handleAction('cancel')"
+            @keydown.enter="handleAction('cancel')"
+          >
+            {{ cancelButtonText || t('el.messagebox.cancel') }}
+          </el-button>
+          <el-button
+            v-show="showConfirmButton"
+            ref="confirm"
+            :loading="confirmButtonLoading"
+            :class="[ confirmButtonClasses ]"
+            :round="roundButton"
+            size="small"
+            @click="handleAction('confirm')"
+            @keydown.enter="handleAction('confirm')"
+          >
+            {{ confirmButtonText || t('el.messagebox.confirm') }}
+          </el-button>
+        </div>
+      </div>
+    </div>
+  </transition>
+</template>
+<script lang='ts'>
+import {
+  defineComponent,
+  nextTick,
+  onMounted,
+  onBeforeUnmount,
+  computed,
+  watch,
+  watchEffect,
+  getCurrentInstance,
+  reactive,
+  toRefs,
+} from 'vue'
+import { addClass, removeClass } from '@element-plus/utils/dom'
+import ElButton from '@element-plus/button/src/button.vue'
+import ElInput from '@element-plus/input/src/index.vue'
+import { t } from '@element-plus/locale'
+import Dialog  from '@element-plus/utils/aria-dialog'
+import usePopup from '@element-plus/utils/popup/usePopup'
+
+let dialog
+
+const TypeMap: Indexable<string> = {
+  success: 'success',
+  info: 'info',
+  warning: 'warning',
+  error: 'error',
+}
+
+export default defineComponent({
+  name: 'ElMessageBox',
+  components: {
+    ElButton,
+    ElInput,
+  },
+  props: Object.assign({}, usePopup.comPropsTypes, {
```

Comment:
terrible code for `props`, why not just use in `setup`

Reply:
how to add props with PropType validate in setup ? or I do like this? it's Ok?
![image](https://user-images.githubusercontent.com/23527768/92696668-c8e27500-f37c-11ea-9898-6a927b16e6c5.png)